### PR TITLE
feat: rework kotel metrics

### PR DESCRIPTION
**Pull request by kruskall**
Remove kotel plugin and publish those metrics in our own hook.
Add compressed/uncompressed fetched bytes.
Use consistent attributes across metrics.

This PR keeps the same name for the kotel metrics for backward comaptibility.

### CODE REVIEW

File: kafka/metrics.go [hunk 1]
```diff
@@ -246,3 +528,10 @@ func attributesFromRecord(r *kgo.Record, extra ...attribute.KeyValue) []attribut
 	}
 	return attrs
 }
+
+func strnode(node int32) string {
```
Comment (endorama) [inline kafka/metrics.go:532]: question: could you clarify why this is needed and why the return value (seems to) differs between `node < 0` and `node >= 0`?

File: kafka/metrics.go [hunk 2]
```diff
@@ -60,11 +70,31 @@ type metricHooks struct {
...
 
-	messageProduced          metric.Int64Counter
-	messageFetched           metric.Int64Counter
-	messageDelay             metric.Float64Histogram
-	messageBytes             metric.Int64Counter
-	messageUncompressedBytes metric.Int64Counter
+	// kotel metrics
+
+	connects    metric.Int64Counter
+	connectErrs metric.Int64Counter
```
Comment (endorama) [inline kafka/metrics.go:76]: question: as we submitted a PR to update these metrics (https://github.com/twmb/franz-go/pull/691) do you think it make sense to keep compatibility with `kotel` here? I don't see benefits in the long term but I'm ok if we want to keep compatibility for this PR

File: kafka/metrics_test.go [hunk 3]
```diff
@@ -386,11 +455,24 @@ func TestConsumerMetrics(t *testing.T) {
 		},
 	}
 
+	ignore := []string{
```
Comment (kyungeunni) [inline kafka/metrics_test.go:458]: Why do we need to ignore them? should we be able to check these metrics too as it's now reported by our own hook?

**Review by kyungeunni — COMMENTED** [review]
Looks great, may I ask what motivated the removal of kotel? Is there a ticket I can refer to have more context?

File: kafka/metrics.go [hunk 4]
```diff
@@ -73,43 +101,168 @@ func newKgoHooks(mp metric.MeterProvider, namespace, topicPrefix string,
 	topicAttributeFunc TopicAttributeFunc,
 ) (*metricHooks, error) {
 	m := mp.Meter(instrumentName)
+
+	// kotel metrics
+
+	// connects and disconnects
+	connects, err := m.Int64Counter(
+		"messaging.kafka.connects.count",
```
Comment (kyungeunni) [inline kafka/metrics.go:109]: Q. Should we define constant values for the kotel metrics, too? On the other hands, it also make sense to inline them to distinguish from our custom metric.

File: kafka/metrics.go (hunk 2 above)
Comment (twmb) [reply kafka/metrics.go:74]: FWIW I plan to merge 691 actually sooner than later, if that's the problem.

File: kafka/metrics.go (hunk 4 above)
Comment (kyungeunni) [reply kafka/metrics.go:109]: Makes sense. thanks for the clarification 👍

**Review by kyungeunni — APPROVED** [review]

**Pull request merged**